app/models/course/settings/notifications.rb

```diff
@@ -32,7 +32,9 @@ def email_settings
     all_settings = settings_interfaces_hash.values.map do |settings|
       settings.respond_to?(:email_settings) ? settings.email_settings : nil
     end
-    all_settings.compact.flatten.sort_by { |item| item[:component] }
+    all_settings.compact.flatten.sort do |a, b|
+      [a[:component], a[:component_title], a[:key]] <=> [b[:component], b[:component_title], b[:key]]
```

Comment:
I happened to be reading about this recently  - I think Ruby's array does implement this sort by default already, so you can do: `all_settings.compact.flatten.sort_by { |item| [item[:component], item[:component_title], item[:key]] }`. 

Perhaps you can try and verify?

Reply:
Yup that works. Edited.